src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,166 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    
+    private int size;
+    private Node first;
+    private Node last;
+    
     @Override
     public void add(T value) {
+        linkLast(value);
     }
-
+    
     @Override
     public void add(T value, int index) {
+        if (!(index >= 0 && index <= size)) {
+            throw new IndexOutOfBoundsException("index is out of array " + index);
+        }
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, node(index));
+        }
     }
-
+    
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
-
+    
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        return node(index).item;
     }
-
+    
+    private void checkElementIndex(int index) {
```

Comment:
```suggestion
    private void checkIndex(int index) {
```
put private method after public ones